src/contractor/contractor.cpp

```diff
@@ -103,22 +103,14 @@ int Contractor::Run()
 
 void Contractor::ReadNodeLevels(std::vector<float> &node_levels) const
 {
-    storage::io::FileReader order_file(config.level_output_path,
-                                       storage::io::FileReader::HasNoFingerprint);
-
-    const auto level_size = order_file.ReadElementCount32();
-    node_levels.resize(level_size);
-    order_file.ReadInto(node_levels);
+    files::readLevels(config.level_output_path, node_levels);
```

Comment:
Thanks for the refactor! I think we can remove `Contractor::ReadNodeLevels` entirely now and directly call `files::readLevels`.